app/Livewire/UploadEmployers.php

```
@@ -45,18 +45,13 @@ public function saveFile()
         try {
             $csv = Reader::createFromPath($this->file->getRealPath())->setHeaderOffset(0);
             $header = $csv->getHeader();
-            if(!$this->checkHeaders($header)){
-                $this->dispatch('alert', type: 'error', title: "Invalid CSV file format. Correct header should be: N/O,
-                first_name,
-                last_name,
-                employee_ID,
-                email,
-                position_role,
-                pay_rate", position: 'center', timer: '7500');
+            $checkHeader = $this->checkHeaders($header);
+            if(!$checkHeader['status']){
+                $this->dispatch('alert', type: 'error', title: $checkHeader['response'], position: 'center', timer: '7500');
             }else{
                 $employees = iterator_to_array($csv, true);
 
-                CreateEmployeeJob::dispatch($employees, $this->company_id, 'file');
+                CreateEmployeeJob::dispatch($employees, $this->company_id);
                 $this->reset();
                 $this->dispatch('alert', type: 'success', title: 'New Employees Records Successfully Added To The Queue For Processing', position: 'center', timer: '2500');
                 return redirect('dasboard')->route('company.dashboard');
```

`redirect('dasboard')` instead of redirect('dashboard')

---

Please throw the exception since we are catching `throw $e`

---

I understand throwing the exception from the queue job, but logging would have been better for the livewire controllers as it will not be too nice for the users to be getting redirected to an error screen